### PR TITLE
Deserialize user fn response before sending response to client

### PR DESCRIPTION
**Pull request by drowzy**
* General improvements to `handle_stream` / `handle_unary`
* Abstracts common logic to Protocol mod
* Deserialize `Any` types returned from user functions

### CODE REVIEW

File: apps/massa_proxy/lib/massa_proxy/runtime/grpc/protocol/action/protocol.ex [hunk 1]
```diff
@@ -0,0 +1,67 @@
+defmodule MassaProxy.Runtime.Grpc.Protocol.Action.Protocol do
```
Comment (drowzy) [inline apps/massa_proxy/lib/massa_proxy/runtime/grpc/protocol/action/protocol.ex:1]: The module name is kind of meh, easy to confuse it with something else. Suggestions for a better name?

Comment (sleipnir) [reply apps/massa_proxy/lib/massa_proxy/runtime/grpc/protocol/action/protocol.ex:1]: remove last Protocol or join with Action like ActionProtocol helps?

File: apps/massa_proxy/lib/massa_proxy/runtime/grpc/protocol/action/stream_handler.ex [hunk 2]
```diff
@@ -3,58 +3,49 @@ defmodule MassaProxy.Runtime.Grpc.Protocol.Action.Stream.Handler do
...
          :ok <- client_stream |> run_stream(messages) |> Stream.run(),
          {:ok, consumer_stream} <- GRPC.Stub.recv(client_stream) do
       consumer_stream
-      |> Stream.each(fn {:ok, r} -> GRPC.Server.send_reply(stream, unpack_response(r)) end)
+      |> Stream.each(fn {:ok, r} ->
+        GRPC.Server.send_reply(stream, ActionProtocol.decode(ctx, r))
+      end)
       |> Stream.run()
     else
       {:error, _reason} = err -> err
```
Comment (sleipnir) [inline apps/massa_proxy/lib/massa_proxy/runtime/grpc/protocol/action/stream_handler.ex:24]: You don't have to worry about that right now but I think we can do better here. We need to think about what would be the best mistake to generate an adequate response for the client in cases of failure.

**Review by sleipnir — APPROVED** [review]
Beautiful work here

**Pull request merged**